File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if len(nums) == 1:return 1 
        
        dp = [1 for _ in range(len(nums))]

        for i in range(1,len(nums)):
            for j in range(0,i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i],dp[j]+1)
            
        return max(dp)

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int[] dp = new int[nums.length];
        for (int i = 0 ; i < nums.length;i++){
            dp[i] = 1;
        }
        for (int i = 0 ; i < nums.length;i++){
            for (int j = i - 1 ; j >= 0 ; j -- ){
                if (nums[i] > nums[j]){
                    dp[i] = Math.max(dp[i],dp[j] + 1);
                }
            }
        }
        int ans = 0;
        for (int i = 0 ; i < nums.length;i++){
            ans = Math.max(dp[i],ans);
        }
        return ans;
        
    }
}